inst/pages/integrated_learner.qmd

```diff
@@ -19,518 +17,161 @@ advanced method, called cooperative learning [@Ding2022], which is also known as
+```{r valid-set}
 
 ###########################
-# Create Species Features #
-###########################
-
-feature_species <- as.data.frame(assay(se_relative))
-rownames(feature_species) <- sub('.*s__', '', rownames(feature_species))
-
+# load required libraries #
 ###########################
-# Create Pathway Features #
-###########################
-
-feature_pwys <- as.data.frame(assay(se_pathway))
-feature_pwys <- rownames_to_column(feature_pwys, "ID")
-feature_pwys <- feature_pwys %>%
-  filter(!grepl("\\|", ID)) %>%
-  filter(!ID %in% c('UNMAPPED', 'UNINTEGRATED')) %>%
-  column_to_rownames('ID') %>%
-  as.data.frame()
-
-################################
-# Combine Species and Pathways #
-################################
-
-feature_table <- bind_rows(feature_species, feature_pwys)
-
-########################################################
-# Check row names of feature_table and sample_metadata #
-########################################################
-
-all(colnames(feature_table) == rownames(sample_metadata))
-
-```
-
-We will then create a metadata table for the features. This table captures
-high-level information related to the features (e.g., which layer they belong
-to).
+library(mia)
+library(SuperLearner)
+library(IntegratedLearner)
+library(tidyverse)
+library(cowplot)
 
-```{r create-metadata-table}
+########################
+# load multi-omic data #
+########################
+data("HintikkaXOData")
+mae <- HintikkaXOData
 
 ######################################
-# Create metadata table for features #
+# Add validation set info to colData #
 ######################################
 
-rowID <- rep(
-    c('Species', 'Pathways'),
-    c(nrow(feature_species), nrow(feature_pwys)))
-feature_metadata <- cbind.data.frame(
-    featureID = rownames(feature_table), featureType = rowID)
-rownames(feature_metadata) <- feature_metadata$featureID
+colData(mae)[["valid"]] <- sample(c(TRUE, FALSE), nrow(colData(mae)), 
+                                  replace = TRUE)
 
-################
-# Sanity check #
-################
+###########################
+# Run with validation set #
+###########################
 
-all(rownames(feature_table) == rownames(feature_metadata)) # TRUE
-all(colnames(feature_table) == rownames(sample_metadata)) # TRUE
+fit1 <- IntegratedLearnerFromMAE(
+    mae,
+    experiment = c(1, 2, 3),
+    assay.type = c("counts", "nmr", "signals"),
+    outcome.col = "Fat",
+    valid.col = "valid",
+    folds = 10,
+    na.rm = TRUE,
+    base_learner = 'SL.randomForest',
+    meta_learner = 'SL.nnls.auc',
+    verbose = TRUE,
+    family = binomial()
+)
 
 ```
 
-Further data pre-processing is necessary to handle near-zero-variance features
-in this dataset. Ultimately, 483 features are retained, consisting of 360
-pathways and 123 species. A combination of variance and prevalence filtering
-is applied to the feature table, while related metadata is cleaned to ensure the
-retention of matching samples.
```

Comment:
Some of these removed texts include good info. Consider keeping them and incorporating them to new version